### ISSUE TITLE
Implement fold detectors for other languages

### ISSUE THREAD

**Issue by rlaverde** [spyder-ide/spyder#4432]
At the moment there is only a generic fold detector based in indentation.

One of the things missing after #3833

**Comment by goanpeca** [comment]
I think we should close this, as now this functionality is deferred to language servers  @ccordoba12

**Comment by ccordoba12** [comment]
Agreed.
